tests/datasources/test_chirps.py

```diff
@@ -0,0 +1,329 @@
+        start_year=_START_YEAR,
+        end_year=_END_YEAR,
+        start_month=_START_MONTH,
+        end_month=_END_MONTH,
+        start_day=_START_DAY,
+        end_day=_END_DAY,
+    )
+
+    assert url == (
+        "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/"
+        ".CHIRPS/.v2p0/.monthly/.global/.precipitation/"
+        f"X/%28-3.0%29%282.0%29RANGEEDGES/"
+        f"Y/%286.0%29%283.2%29RANGEEDGES/"
+        f"T/%28{_END_MONTH_STR}%20{_END_YEAR}%29"
+        f"%28{_END_MONTH_STR}%20{_END_YEAR}"
+        "%29RANGEEDGES/data.nc"
+    )
+
+    assert (
+        filepath
+        == mock_aa_data_dir
+        / f"public/raw/{mock_country_config.iso3}/{DATASOURCE_BASE_DIR}/"
+        f"abc_chirps_monthly_{_END_YEAR}_{_END_MONTH:02d}_"
+        "r0.05_Np6Sp3Ep2Wm3.nc"
+    )
+
+
+def test_download_daily(mock_aa_data_dir, mock_country_config, mock_download):
+    # TODO: change docstrings
+    """Test that the correct country url and path is returned."""
+    url, filepath = mock_download(
+        frequency="daily",
+        start_year=_START_YEAR,
+        end_year=_END_YEAR,
+        start_month=_START_MONTH,
+        end_month=_END_MONTH,
+        start_day=_START_DAY,
+        end_day=_END_DAY,
+    )
+
+    assert url == (
+        "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/"
+        ".CHIRPS/.v2p0/.daily-improved/.global/.0p05/.prcp/"
+        f"X/%28-3.0%29%282.0%29RANGEEDGES/"
+        f"Y/%286.0%29%283.2%29RANGEEDGES/"
+        f"T/%28{_END_DAY:02d}%20{_END_MONTH_STR}%20{_END_YEAR}"
+        f"%29%28{_END_DAY:02d}%20{_END_MONTH_STR}%20{_END_YEAR}"
+        "%29RANGEEDGES/data.nc"
+    )
+
+    assert (
+        filepath
+        == mock_aa_data_dir
+        / f"public/raw/{mock_country_config.iso3}/{DATASOURCE_BASE_DIR}/"
+        f"abc_chirps_daily_{_END_YEAR}_{_END_MONTH:02d}_{_END_DAY:02d}_"
+        "r0.05_Np6Sp3Ep2Wm3.nc"
+    )
+
+
+def test_create_date_list_daily(mock_chirps):
+    """Test the creation of the date list for daily data."""
+    chirps = mock_chirps(frequency="daily")
+    date_list = [("2020", "07", f"{day:02d}") for day in range(_START_DAY, 32)]
+    date_list.append(("2020", "08", "01"))
+
+    assert (
+        date_list
+        == chirps._create_date_list(
+            start_year=_START_YEAR,
+            end_year=_END_YEAR,
+            start_month=_START_MONTH,
+            end_month=_END_MONTH,
+            start_day=_START_DAY,
+            end_day=_END_DAY,
+        )[0]
+    )
+
+
+def test_create_date_list_monthly(mock_chirps):
+    """Test the creation of the date list for monthly data."""
+    chirps = mock_chirps(frequency="monthly")
+    date_list = [("2020", "07"), ("2020", "08")]
+
+    assert (
+        date_list
+        == chirps._create_date_list(
+            start_year=_START_YEAR,
+            end_year=_END_YEAR,
+            start_month=_START_MONTH,
+            end_month=_END_MONTH,
+            start_day=_START_DAY,
+            end_day=_END_DAY,
+        )[0]
+    )
+
```

Comment:
I know this becomes a lot of testing but you might also want to test when some of the input args to `_create_date_list` are None